app/models/mixins/aggregation_mixin/methods.rb

```diff
@@ -43,7 +43,7 @@ def aggregate_hardware(from, field, targets = nil)
       select    = field == :aggregate_cpu_speed ? "cpu_total_cores, cpu_speed" : field
       targets ||= send("all_#{from}_ids")
       targets   = targets.collect(&:id) unless targets.first.kind_of?(Integer)
-      hdws      = Hardware.where("#{from}_id".to_sym => targets).select(select)
+      hdws      = Hardware.where(from.to_sym => targets).select(select)
```

Comment:
This actually changes the query from being on the ids column to being on I guess the association?  (are there specs around this method?)

Reply:
Oh wow this actually works passing ids without specifying _id... TIL

```ruby
[2] pry(main)> MiqServer.where(:zone_id => [10000000000002]).count
   (3.5ms)  SELECT COUNT(*) FROM "miq_servers" WHERE "miq_servers"."zone_id" = $1  [["zone_id", 10000000000002]]
=> 2
[3] pry(main)> MiqServer.where(:zone => [10000000000002]).count
   (0.4ms)  SELECT COUNT(*) FROM "miq_servers" WHERE "miq_servers"."zone_id" = $1  [["zone_id", 10000000000002]]
=> 2
```